cloudify_agent/installer/script.py

```diff
@@ -109,39 +110,38 @@ def _get_script_path_and_url(self, script_filename):
         script_url = url_join(self.file_server_url, script_relpath)
         return script_path, script_url
 
-    def _script_download_link(self, is_install_script=True):
-        if is_install_script:
-            script_filename = self.install_script_filename
-            script_content = self.install_script()
-        else:
-            script_filename = self.init_script_filename
-            script_content = self.init_script()
-
-        script_path, script_url = self._get_script_path_and_url(
-            script_filename
-        )
-        with open(script_path, 'w') as script_file:
-            script_file.write(script_content)
-
-        return script_path, script_url
-
-    def install_script_download_link(self):
-        """Get agent installation script and write it to file server location.
+    def init_script_download_link(self):
+        """Get agent init script and write it to file server location.
         :return: A tuple with:
         1. Path where the install script resides in the file server
         2. URL where the install script can be downloaded
         :rtype: (str, str)
         """
-        return self._script_download_link(is_install_script=True)
+        script_filename = self.init_script_filename
+        script_content = self.init_script()
+        return self._get_script_path_and_url(script_filename, script_content)
 
-    def init_script_download_link(self):
-        """Get agent init script and write it to file server location.
+    def install_script_download_link(self, add_ssl_cert=True):
+        """Get agent installation script and write it to file server location.
         :return: A tuple with:
         1. Path where the install script resides in the file server
         2. URL where the install script can be downloaded
         :rtype: (str, str)
         """
-        return self._script_download_link(is_install_script=False)
+        script_filename = self.install_script_filename
+        script_content = self.install_script(add_ssl_cert=add_ssl_cert)
+        return self._get_script_path_and_url(script_filename, script_content)
+
+    def _get_script_path_and_url(self, script_filename, script_content):
```

Comment:
Sorry, but it would be  really difficult to separate the changes in this file into more than one coherent commit